WyattHWilliams/react-challenge-project-jan-2022
# Feat 8 edit & delete

## Changes
1. created a deleteOrder function.
2. created an /edit-order/:id route for editing specific orders using the already made order-form component
3. Changed the edit button in the ordersList component to a link to the /edit-order/:id route that passes the order as persisted state to the order-form component.
4. Modified the order-form component to include functionality for editing the order, and for checking to see if a new order is being made or if an old order is being changed

## Purpose
We wanted to add delete and edit functionality without creating any new components and without adding a whole bunch of code to keep it as "dry" as possible.

## Approach
1. By reusing a slightly modified order-form component we avoid making any new components related to editing an order.
2. By using an edit-order/:id path we separate concerns
3. By using a link to the edit-order path we can pass the order as state directly to the order-form component without that component needing to be a child of OrdersList, and we get to persist the data for that order without adding to, or creating, a reducer.

## Testing Steps
for Deleting an order:
1. create an order, see that it is created properly and renders in the view orders list
2. click the delete button
3. Check that it is no longer rendered
4. call for that order's id again from the backend and check that it does not exist

## Learning
I consulted these two articles to find a way to pass state from one non-related component to another inside the router:
https://www.daggala.com/passing-props-through-link-in-react-router/
https://ui.dev/react-router-pass-props-to-link/

I initially thought about just passing the order id through the url and using useParams to get the id and make a GET call to the backend, but I decided no to for 2 reasons:
1. would have been more code, and I wanted to keep it as "dry" as possible
2. The backend has no route for getting individual orders

Closes #008
